test/Enable.Extensions.Queuing.AzureServiceBus.Tests/ConcurrentQueueClientTests.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.Security;
+using Enable.Extensions.Queuing.AzureServiceBus.Internal;
+using Xunit;
+
+namespace Enable.Extensions.Queuing.AzureServiceBus.Tests
+{
+    public class ConcurrentQueueClientTests
+    {
+        private readonly AzureServiceBusQueueClientFactory _factory;
+        private readonly string _queueName;
+
+        public ConcurrentQueueClientTests()
+        {
+            var connectionString = GetEnvironmentVariable("AZURE_SERVICE_BUS_CONNECTION_STRING");
+
+            var factoryOptions = new AzureServiceBusQueueClientFactoryOptions
+            {
+                ConnectionString = connectionString
+            };
+
+            _factory = new AzureServiceBusQueueClientFactory(factoryOptions);
+            _queueName = GetEnvironmentVariable("AZURE_SERVICE_BUS_QUEUE_NAME");
+        }
+
+        [Fact]
+        public void ConcurrentQueueClientsUseSameQueue()
+        {
+            // Arrange
+            var client = _factory.GetQueueReference(_queueName) as AzureServiceBusQueueClient;
+            var anotherClient = _factory.GetQueueReference(_queueName) as AzureServiceBusQueueClient;
+
+            // Assert
+            Assert.True(object.ReferenceEquals(client.Queue, anotherClient.Queue));
+
+            // Clean up
+            client.Dispose();
+            anotherClient.Dispose();
+        }
+
+        [Fact]
+        public void UnconcurrentQueueClientsUseDifferentQueue()
+        {
+            // Arrange
+            var client = _factory.GetQueueReference(_queueName) as AzureServiceBusQueueClient;
+            client.Dispose(); // Dispose of this client before we make another.
+
+            var anotherClient = _factory.GetQueueReference(_queueName) as AzureServiceBusQueueClient;
+
+            // Assert
+            Assert.False(object.ReferenceEquals(client.Queue, anotherClient.Queue));
```

Comment:
The tests in this class appear to be testing that some internal property shared between instances of queue clients. However, this isn't testing that a single connection to Service Bus is shared between these instances. Is there a more appropriate test that we can use instead?

Reply:
See comment [here](https://github.com/EnableSoftware/Enable.Extensions.Queuing/pull/28#issuecomment-471521347) and [latest code](666dab50be8bc06913c2ddc9127499345b6cbbed).